### PR TITLE
Use CONNECTION_CLOSE rather than CRYPTO_CLOSE

### PR DESCRIPTION
**Pull request by martinthomson**
More frame types aren't really necessary here.

(This will conflict with the rename of CRYPTO_HS, but I can manage that.)

### CODE REVIEW

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -602,10 +602,16 @@ HelloRetryRequest is still used for incorrect key shares.
...
 
-If TLS experiences an error, it MUST generate an appropriate alert
-as defined in {{TLS13}}; Section 6) and then provide it to QUIC,
-which sends the alert in a CRYPTO_CLOSE frame. All such alerts are
-"fatal" (see {{TLS13}}, Section 6.2.
+If TLS experiences an error, it generates an appropriate alert as defined in
+Section 6 of {{TLS13}}.
+
+A TLS alert is turned into a QUIC connection error by converting the alert
+description into a QUIC error code.  The alert description is added to 0x200 to
```
Comment (ekr) [inline draft-ietf-quic-tls.md]: Probably worth observing that this is a one-byte field.

File: draft-ietf-quic-tls.md [hunk 2]
```diff
@@ -602,10 +602,16 @@ HelloRetryRequest is still used for incorrect key shares.
...
+produce a QUIC error code from the range reserved for CRYPTO_ERROR.  The
+resulting value is sent in a QUIC CONNECTION_CLOSE frame.
+
+The alert level of all TLS alerts is "fatal", a TLS stack does not generate
+alerts at the "warning" level.
```
Comment (ekr) [inline draft-ietf-quic-tls.md]: This is a bit confusing because stacks can in fact do that. Maybe MUST NOT?

Comment (martinthomson) [reply draft-ietf-quic-tls.md]: I thought that this was OK.  TLS 1.3 prohibits use of warning for anything but close_notify, and we don't allow TLS to close.  I guess that MUST NOT isn't a problem though.

**Comment by rpaulo** [comment]
I really like this simplification.

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -4150,45 +4133,40 @@ The most appropriate error code ({{error-codes}}) SHOULD be included in the
...
 transport, including all those described in this document, are carried
 in a CONNECTION_CLOSE frame.  Other than the type of error code they
 carry, these frames are identical in format and semantics.
 
-A CONNECTION_CLOSE, CRYPTO_CLOSE, or APPLICATION_CLOSE frame could be
-sent in a packet that is lost.  An endpoint SHOULD be prepared to
-retransmit a packet containing either frame type if it receives more
-packets on a terminated connection.  Limiting the number of
-retransmissions and the time over which this final packet is sent
+A CONNECTION_CLOSE, or APPLICATION_CLOSE frame could be sent in a packet that is
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: "," unnecessary

File: draft-ietf-quic-tls.md [hunk 4]
```diff
@@ -602,10 +602,16 @@ HelloRetryRequest is still used for incorrect key shares.
...
-"fatal" (see {{TLS13}}, Section 6.2.
+If TLS experiences an error, it generates an appropriate alert as defined in
+Section 6 of {{TLS13}}.
+
+A TLS alert is turned into a QUIC connection error by converting the one-octet
+alert description into a QUIC error code.  The alert description is added to
+0x200 to produce a QUIC error code from the range reserved for CRYPTO_ERROR.
+The resulting value is sent in a QUIC CONNECTION_CLOSE frame.
+
+The alert level of all TLS alerts is "fatal", a TLS stack MUST NOT generate
```
Comment (MikeBishop) [inline draft-ietf-quic-tls.md]: Comma splice.  Separate sentences or `;`.

File: draft-ietf-quic-transport.md [hunk 5]
```diff
@@ -1032,7 +1032,6 @@ explained in more detail as they are referenced later in the document.
 | 0x0f        | PATH_RESPONSE     | {{frame-path-response}}     |
 | 0x10 - 0x17 | STREAM            | {{frame-stream}}            |
 | 0x18        | CRYPTO_HS         | {{frame-crypto}}            |
-| 0x19        | CRYPTO_CLOSE      | {{frame-crypto-close}}      |
 | 0x20        | NEW_TOKEN         | {{frame-new-token}}         |
```
[the remark this answers is not in the document]
Comment (MikeBishop) [reply draft-ietf-quic-transport.md]: No objection, but also seems unnecessary.  Leave 0x19 for the next frame to be defined, and if there are gaps in the space we'll defrag before we ship.

File: draft-ietf-quic-transport.md [hunk 6]
```diff
@@ -2200,10 +2199,9 @@ before the packet is received.
 
 ### Immediate Close
 
-An endpoint sends a closing frame, (CONNECTION_CLOSE, CRYPTO_CLOSE or
-APPLICATION_CLOSE) to terminate the connection immediately.  Any
-closing frame causes all streams to immediately become closed; open
-streams can be assumed to be implicitly reset.
+An endpoint sends a closing frame, (CONNECTION_CLOSE or APPLICATION_CLOSE) to
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md]: Unnecessary comma

**Pull request merged**